src/librustc/hir/lowering.rs

```diff
@@ -101,8 +108,8 @@ impl<'a, 'hir> LoweringContext<'a> {
         LoweringContext {
             crate_root: None,
             id_assigner: id_assigner,
-            definitions: None,
             parent_def: Cell::new(None),
+            resolver: None,
```

Comment:
Rather than using an `Option` for resolver, could you have a dummy resolver which implements `Resolver` and silently ignores requests to resolve?